python/dgl/distributed/partition.py

```diff
@@ -24,6 +34,43 @@
     ETYPE: F.int32
     }
 
+def _format_part_metadata(part_metadata, formator):
+    ''' Format etypes with specified formator
+    '''
+    for key in ['edge_map', 'etypes']:
+        if key not in part_metadata:
+            continue
+        orig_data = part_metadata[key]
+        if not isinstance(orig_data, dict):
+            continue
+        new_data = {}
+        for etype, data in orig_data.items():
+            etype = formator(etype)
+            new_data[etype] = data
+        part_metadata[key] = new_data
+    return part_metadata
+
+def _load_part_config(part_config):
+    ''' Load part config and format
+    '''
+    try:
+        with open(part_config) as f:
+            part_metadata = _format_part_metadata(json.load(f),
+                _etype_str_to_tuple)
+    except AssertionError as e:
+        raise DGLError(f"Failed to load partition config due to {e}. "
```

Comment:
I saw sometimes we are using assertion but sometimes raise error. Do we have a coherent suggestion? @frozenbugs

Reply:
The key is to provide enough information when error happens. Solely using assert is not so good, the error msg will just be, assert failure.

But we can easily create a wrapper assert function with a string as input, such that we can simplify the code structure.

Reply:
Probably worth looking at how other python package do logging/checking. For C++, we use GLOG.

Reply:
For me, when I need to throw an exception and to be caught by caller, I choose to raise DGLError. If I don't want to catch it, just let the exception aborts the app, use `assert` is ok for me.